src/battle-scene.ts

```diff
@@ -1373,7 +1374,16 @@ export default class BattleScene extends SceneBase {
 
         for (const pokemon of this.getPlayerParty()) {
           pokemon.resetWaveData();
+          pokemon.resetTera(); // TODO: put this in resetBattleData?
+
           applyAbAttrs<PostBattleInitAbAttr>(AbAttrFlag.POST_BATTLE_INIT, pokemon, false);
+
+          if (
+            pokemon.species.speciesId === SpeciesId.TERAPAGOS
+            || (this.gameMode.isClassic && isBetween(this.currentBattle.waveIndex, 181, 190))
+          ) {
+            this.arena.playerTerasUsed = 0;
+          }
```

Comment:
I assume this is this:
> Upon capturing Terapagos at the end of [The Indigo Disk](https://bulbapedia.bulbagarden.net/wiki/The_Indigo_Disk), the Tera Orb will be infused with Terapagos' energy and will no longer require charging.

We should document that in a TLDR way

```suggestion

          // Upon capturing TERAPAGOS [...],  the Tera Orb will be infused with Terapagos' energy and will no longer require charging
          if (
            pokemon.species.speciesId === SpeciesId.TERAPAGOS
            || (this.gameMode.isClassic && isBetween(this.currentBattle.waveIndex, 181, 190))
          ) {
            this.arena.playerTerasUsed = 0;
          }
```

Although it seems that here the player hasn't caught it so not sure

Reply:
Would also add on to that comment about how it's automatically recharged from 181-190